README.md

```diff
@@ -1,2 +1,52 @@
 # Graph Compiler
-Graph Compiler is in active development stage.
\ No newline at end of file
+Graph Compiler is in active development stage.
+
+## Build instructions
+
+All-on-one compile script is at `scripts/compile.sh`.
+
+To build this project step by step, first you need to find the LLVM commit-id we are using at `cmake/llvm-version.txt`. Then clone specific version of LLVM:
+
+```bash
+LLVM_COMMIT=????? # the commit id in cmake/llvm-version.txt of this repo
+git clone https://github.com/llvm/llvm-project
+cd llvm-project
+git checkout $LLVM_COMMIT
+```
+
+Build LLVM with the command lines given in `.github/workflows/build-llvm.yml`:
+
+```bash
+mkdir llvm-install
+cmake -G Ninja llvm -B build -DCMAKE_INSTALL_PREFIX=llvm-install \
+  -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=true -DLLVM_ENABLE_PROJECTS="mlir" -DLLVM_TARGETS_TO_BUILD="X86" -DLLVM_INSTALL_UTILS=true -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DLLVM_INSTALL_GTEST=ON
```

Comment:
Suggest to build and link as dynamic lib to reduce the overall size?
BTW, is there any difference for `ON` and `true`? I thought `ON` is more widely used than the other one?

Reply:
Looks like they have the same meaning?

https://enccs.github.io/cmake-workshop/cmake-syntax/

```
True is any expression evaluating to: 1, ON, TRUE, YES, and Y.
False is any expression evaluating to: 0, OFF, FALSE, NO, N, IGNORE, and NOTFOUND.
```